fs/layer/layer.go

```diff
@@ -345,7 +345,11 @@ func (r *Resolver) Resolve(ctx context.Context, hosts []docker.RegistryHost, ref
 	if err != nil {
 		return nil, fmt.Errorf("failed to read layer: %w", err)
 	}
-	disableXattrs := getDisableXAttrAnnotation(sociDesc)
+	disableXattrs := false
+	if !r.config.ForceGetXAttrs {
```

Comment:
nitpick but this reads a little strange to me, as opposed to

```
disableXattrs := true
if r.config.ForceGetXAttrs {
```

feels a bit odd that the "default behavior" is to enter the if statement

Reply:
The way I see it is, by default, we're checking for xattrs. But, if our config does not tell us that we are required to do this, check the annotation to see if we can maybe skip this step.

It's kind of confusing, though, I agree. I just pushed a change to this, LMK if it's better. (It's technically a tiny bit less optimal but probably inconsequential)

TBH I think the var name is just bad :p If it were up to me, it would be something like "checkXAttrs" and pass that down the call chain. But since FUSE names this variable [DisableXAttrs](https://github.com/hanwen/go-fuse/blob/36b359112698c1ccc5d5139693b92f62eb02477c/fuse/api.go#L219) it would probably be more confusing to deviate from the name.